azkaban-web-server/src/main/java/azkaban/webapp/servlet/ExecutorServlet.java

```diff
@@ -883,6 +883,17 @@ private void ajaxAttemptExecuteFlow(final HttpServletRequest req,
     final String projectName = getParam(req, "project");
     final String flowId = getParam(req, "flow");
 
+    // check proxy
+    final  Map<String,String> props = getParamGroup(req, "flowOverride");
+    if (props != null){
+      for (String key : props.keySet()){
+         if ("user.to.proxy".equals(key) && !userManager.validateProxyUser(props.get(key), user)){
```

Comment:
Hi @sthbig , I believe we have this check when job starts to run. What's the advantage having this check here?

Reply:
Hi @kunkun-tang , thanks for reply. Actually, there is no proxy check before executing flow. And the bug can be reproduced.

Following are my steps:
1.Before doing an experiment, I created a user who only has read permission. The conf/azkaban-users.xml looks like this.
```
<azkaban-users>
  <user groups="azkaban" password="azkaban" roles="admin" username="azkaban"/>
  <user password="metrics" roles="metrics" username="metrics"/>
  <user password="read" roles="read" username="read"/>

  <role name="admin" permissions="ADMIN"/>
  <role name="metrics" permissions="METRICS"/>
  <role name="read" permissions="read"/>
</azkaban-users>
```
2.start Azkaban-web-server and login in as read.

3.create a project and execute flow. Before executing flow, set user.to.proxy property to azkaban. 

4.Azkaban completes the job successfully.
![azkaban web client 2018-03-19 15-57-50](https://user-images.githubusercontent.com/23277484/37584358-2129dd00-2b90-11e8-9060-e6f4297e3ec5.png)
![azkaban web client 2018-03-19 15-58-55](https://user-images.githubusercontent.com/23277484/37584359-214bcab4-2b90-11e8-9dcb-2ddbc846cc37.png)